src/emit/binary.rs

```diff
@@ -829,7 +829,13 @@ impl<'a> Binary<'a> {
                         &ns.structs[*n]
                             .fields
                             .iter()
-                            .map(|f| self.llvm_field_ty(&f.ty, ns))
+                            .map(|f| {
+                                if f.ty.is_mapping() {
+                                    self.llvm_type(&ns.storage_type(), ns)
+                                } else {
+                                    self.llvm_field_ty(&f.ty, ns)
+                                }
+                            })
```

Comment:
I'm not sure this is correct.

A struct field could be an array of two mappings, for example.  This won't work with this code change. A mapping is always stored in storage.

`fn llvm_type()` should always returns `self.llvm_type(&ns.storage_type(), ns)` for type mapping, and this if is no longer needed.